File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/KnowledgeBaseApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class KnowledgeBaseApi {
    private final HttpClient client;
    
    public KnowledgeBaseApi(HttpClient client) {
        this.client = client;
    }

    /** List documents */
    public ListDocumentsGetResponse listDocuments(String knowledge_base_id, Map<String, Object> params) throws Exception {
        return (ListDocumentsGetResponse) client.get(ApiPaths.aiPath("/knowledge-bases/" + knowledge_base_id + "/documents"), params);
    }

    /** Add document */
    public DocumentResponse addDocument(String knowledge_base_id, DocumentAddRequest body) throws Exception {
        return (DocumentResponse) client.post(ApiPaths.aiPath("/knowledge-bases/" + knowledge_base_id + "/documents"), body);
    }

    /** Search knowledge */
    public KnowledgeSearchResponse searchKnowledge(KnowledgeSearchRequest body) throws Exception {
        return (KnowledgeSearchResponse) client.post(ApiPaths.aiPath("/knowledge-bases/search"), body);
    }

    /** List knowledge bases */
    public KnowledgeBaseListResponse listKnowledgeBases(Map<String, Object> params) throws Exception {
        return (KnowledgeBaseListResponse) client.get(ApiPaths.aiPath("/knowledge-bases"), params);
    }

    /** Create knowledge base */
    public KnowledgeBaseResponse createKnowledgeBase(KnowledgeBaseCreateRequest body) throws Exception {
        return (KnowledgeBaseResponse) client.post(ApiPaths.aiPath("/knowledge-bases"), body);
    }

    /** Get document */
    public DocumentResponse getDocument(String knowledge_base_id, String document_id) throws Exception {
        return (DocumentResponse) client.get(ApiPaths.aiPath("/knowledge-bases/" + knowledge_base_id + "/documents/" + document_id + ""));
    }

    /** Delete document */
    public Void deleteDocument(String knowledge_base_id, String document_id) throws Exception {
        client.delete(ApiPaths.aiPath("/knowledge-bases/" + knowledge_base_id + "/documents/" + document_id + ""));
        return null;
    }

    /** Get knowledge base */
    public KnowledgeBaseResponse getKnowledgeBase(String knowledge_base_id) throws Exception {
        return (KnowledgeBaseResponse) client.get(ApiPaths.aiPath("/knowledge-bases/" + knowledge_base_id + ""));
    }

    /** Delete knowledge base */
    public Void deleteKnowledgeBase(String knowledge_base_id) throws Exception {
        client.delete(ApiPaths.aiPath("/knowledge-bases/" + knowledge_base_id + ""));
        return null;
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/KnowledgeBaseApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class KnowledgeBaseApi(private val client: HttpClient) {

    /** List documents */
    suspend fun listDocuments(knowledge_base_id: String, params: Map<String, Any>? = null): ListDocumentsGetResponse? {
        return client.get(ApiPaths.aiPath("/knowledge-bases/$knowledge_base_id/documents"), params) as? ListDocumentsGetResponse
    }

    /** Add document */
    suspend fun addDocument(knowledge_base_id: String, body: DocumentAddRequest): DocumentResponse? {
        return client.post(ApiPaths.aiPath("/knowledge-bases/$knowledge_base_id/documents"), body) as? DocumentResponse
    }

    /** Search knowledge */
    suspend fun searchKnowledge(body: KnowledgeSearchRequest): KnowledgeSearchResponse? {
        return client.post(ApiPaths.aiPath("/knowledge-bases/search"), body) as? KnowledgeSearchResponse
    }

    /** List knowledge bases */
    suspend fun listKnowledgeBases(params: Map<String, Any>? = null): KnowledgeBaseListResponse? {
        return client.get(ApiPaths.aiPath("/knowledge-bases"), params) as? KnowledgeBaseListResponse
    }

    /** Create knowledge base */
    suspend fun createKnowledgeBase(body: KnowledgeBaseCreateRequest): KnowledgeBaseResponse? {
        return client.post(ApiPaths.aiPath("/knowledge-bases"), body) as? KnowledgeBaseResponse
    }

    /** Get document */
    suspend fun getDocument(knowledge_base_id: String, document_id: String): DocumentResponse? {
        return client.get(ApiPaths.aiPath("/knowledge-bases/$knowledge_base_id/documents/$document_id")) as? DocumentResponse
    }

    /** Delete document */
    suspend fun deleteDocument(knowledge_base_id: String, document_id: String): Unit {
        client.delete(ApiPaths.aiPath("/knowledge-bases/$knowledge_base_id/documents/$document_id"))
    }

    /** Get knowledge base */
    suspend fun getKnowledgeBase(knowledge_base_id: String): KnowledgeBaseResponse? {
        return client.get(ApiPaths.aiPath("/knowledge-bases/$knowledge_base_id")) as? KnowledgeBaseResponse
    }

    /** Delete knowledge base */
    suspend fun deleteKnowledgeBase(knowledge_base_id: String): Unit {
        client.delete(ApiPaths.aiPath("/knowledge-bases/$knowledge_base_id"))
    }
}
